prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -209,6 +215,42 @@ def __call__(self, summary: str) -> str:
         return summary
 
 
+class GetWalletBalance(Function):
+    @property
+    def description(self) -> str:
+        return "Use this function to fetch your balance, given in xDAI units."
+
+    @property
+    def example_args(self) -> list[str]:
+        return []
+
+    def __call__(self, user_address: str) -> Decimal:
+        # We focus solely on xDAI balance for now to avoid the agent having to wrap/unwrap xDAI.
```

Comment:
Should the focus be on xDai, then? Redeemed winnings will be visible in wxDai.

Reply:
The balance in wxDAI should be added in the future as well - just limiting it to xDAI because it's the token it can spend when making bets. But great point.